gneiss/q2/_type.py

```diff
@@ -0,0 +1,23 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016--, gneiss development team.
+#
+# Distributed under the terms of the GPLv3 License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+from qiime.plugin import SemanticType
+from q2_types.feature_table import (
+    FeatureTable, Frequency, BIOMV100DirFmt, BIOMV210DirFmt)
+
+
+# Regression types
+Regression = SemanticType('Regression', field_names=['type'])
+
+Linear = SemanticType('Linear',
+                      variant_of=Regression.field['type'])
+LinearMixedEffects = SemanticType('LinearMixedEffects',
+                                  variant_of=Regression.field['type'])
+
+# Tree types
+Hierarchy = SemanticType('Hierarchy', field_names=['type'])
+Cluster = SemanticType('Cluster', variant_of=Hierarchy.field['type'])
```

Comment:
@jairideout @gregcaporaso this was the original motivation for raising https://github.com/qiime2/q2-types/issues/99

While there are some scenarios that can use a phylogeny for an analysis, most scenarios will deal with different types of hierarchies, whether it be generated from a clustering or something else.

I see a few options for this

1. Define a completely orthogonal type to phylogeny named `Hierarchy`
2. Make `Hierarchy` a super type

Also, would we have to worry about clashing types?  For example, if I have defined `Hierarchy[Cluster]` here, could that potentially clash with an unsupervised sampling clustering type implemented in a classifier package?